nupd/fetchers/github.py

```diff
@@ -67,6 +126,59 @@ def get_prefetch_url(self) -> str:
         return f"{self.url}/archive/{self.commit}.tar.gz"
 
 
+async def get_latest_release(
+    owner: str, repo: str, github_token: str | None = None
+) -> GitHubRelease | None:
+    """Get information about the latest release."""
+    session = inject.instance(aiohttp.ClientSession)
+    url = f"https://api.github.com/repos/{owner}/{repo}/releases/latest"
+
+    headers = {
+        "Accept": "application/vnd.github+json",
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+    if github_token:
+        headers["Authorization"] = f"Bearer {github_token}"
+
+    async with session.get(url, headers=headers) as response:
+        if response.status == 404:
+            return None
+        response.raise_for_status()
+        data = await response.json()
+        return GitHubRelease(
+            name=data.get("name"),
+            tag_name=data["tag_name"],
+            committed_at=data["created_at"],
+        )
+
+
+async def get_tags(
+    owner: str, repo: str, github_token: str | None = None
+) -> c.Iterable[GitHubTag] | None:
+    """Get information about a specific release by tag."""
+    session = inject.instance(aiohttp.ClientSession)
+    url = f"https://api.github.com/repos/{owner}/{repo}/tags"
+
+    headers = {
+        "Accept": "application/vnd.github+json",
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+    if github_token:
+        headers["Authorization"] = f"Bearer {github_token}"
+
+    async with session.get(url, headers=headers) as response:
+        if response.status == 404:
+            return None
+        response.raise_for_status()
+        data = await response.json()
+        return [
+            GitHubTag(
+                name=tag_data["name"], commit_sha=tag_data["commit"]["sha"]
+            )
+            for tag_data in data
+        ]
```

Comment:
For better performance, we could do

```suggestion
        for tag_data in data:
            yield GitHubTag(
                name=tag_data["name"], commit_sha=tag_data["commit"]["sha"]
            )
```